kubernetes_env/benchmark/benchmark.py

```diff
@@ -139,73 +184,103 @@ def post_request(_):
     return output
 
 
+def build_and_deploy_filter(filter_dir):
+    res = kube_env.build_filter(filter_dir)
+
+    if res != util.EXIT_SUCCESS:
+        log.error(
+            "Building filter failed for %s."
+            " Make sure you give the right path", filter_dir)
+        return util.EXIT_FAILURE
+
+    res = kube_env.refresh_filter(filter_dir)
+    if res != util.EXIT_SUCCESS:
+        log.error(
+            "Deploying filter failed for %s."
+            " Make sure you give the right path", filter_dir)
+        return util.EXIT_FAILURE
+
+    # wait for kubernetes set up to finish
+    time.sleep(120)
+    return util.EXIT_SUCCESS
+
+
 def start_benchmark(filter_dirs, platform, threads, qps, run_time, **kwargs):
     if kube_env.check_kubernetes_status() != util.EXIT_SUCCESS:
         log.error("Kubernetes is not set up."
                   " Did you run the deployment script?")
         return util.EXIT_FAILURE
+
+    custom = kwargs.get("custom")
+    request = kwargs.get("request")
+    output = kwargs.get("output")
+    command_args = " ".join(kwargs.get("command_args"))
+    application = APPLICATIONS.get(kwargs.get("application"))
+
     _, _, gateway_url = kube_env.get_gateway_info(platform)
     path = kwargs.get("subpath")
     url = f"http://{gateway_url}/{path}"
     log.info("Gateway URL: %s", url)
+
     results = []
     filters = []
+
     if kwargs.get("no_filter") == "ON":
         filter_dirs.append("no_filter")
         filters.append("no_filter")
```

Comment:
This is something I changed locally but forgot to mention - if no_filter is at the end of filter_dirs, then the filter isn't automatically lifted before we run no_filter.  So instead of appending, we should make sure it's the first one in the list, so it can run before any filter gets deployed.

Reply:
Oh dang, I totally missed it. Thanks for catching that!